### PR TITLE
Get current block fix

### PR DESCRIPTION
**Pull request by Dalyj24**
This addresses the duplication of tags discovered while addressing break statements

### CODE REVIEW

File: src/ir/block.rs
Comment (CalebLItalien) [inline src/ir/block.rs:1]: Nice work! I like this function, it almost certainly works. Can you add tests for this (unless they already exist)?

File: src/memory_management/pointer.rs
Comment (CalebLItalien) [inline src/memory_management/pointer.rs:1]: Why do we need to add `Clone` to `CPointer`? Does this affect our memory safety in any way?

File: src/memory_management/resource_pools.rs
Comment (CalebLItalien) [inline src/memory_management/resource_pools.rs:1]: I don't think we should expose a public API for creating tags, unless we absolutely must. What are we using this for?

Comment (CalebLItalien) [inline src/memory_management/resource_pools.rs:1]: Can you change this to be wrapped in an `Option` enum, then initialized to `None`?

Comment (CalebLItalien) [inline src/memory_management/resource_pools.rs:1]: You're able to hash LLVM pointers like this? I'm surprised its not stopping us from trying to do this.

**Review by CalebLItalien — COMMENTED** [review]
Nice work! I like your solution. A few comments/inquiries before we merge this.

File: src/ir/block.rs
Comment (Dalyj24) [reply src/ir/block.rs:1]: Added!

File: src/memory_management/resource_pools.rs
Comment (Dalyj24) [reply src/memory_management/resource_pools.rs:1]: I'm not sure what you mean here, I don't think I exposed anything that wasn't already exposed

Comment (Dalyj24) [reply src/memory_management/resource_pools.rs:1]: so am I but it seems to work so I guess they just decay to their number representations

Comment (Dalyj24) [reply src/memory_management/resource_pools.rs:1]: Done

Comment (Dalyj24) [reply src/memory_management/resource_pools.rs:1]: Fixed!

File: src/memory_management/pointer.rs
Comment (CalebLItalien) [inline src/memory_management/pointer.rs:1]: Can we remove the derivation of debug too? We should try and derive only what is strictly necessary for `CPointer`.

File: src/memory_management/resource_pools.rs
Comment (CalebLItalien) [inline src/memory_management/resource_pools.rs:1]: This is fine, but there is syntactic sugar for creating a vector if it doesn't exist. Look at the store functions on how to do this, if you want to.

File: tests/ir_tests/block_tests.rs
Comment (CalebLItalien) [inline tests/ir_tests/block_tests.rs:1]: Nice test!

**Review by CalebLItalien — COMMENTED** [review]
Excellent work! A few notes before we merge this.

**Review by CalebLItalien — APPROVED** [review]
Nice work!

**Pull request merged**